src/engine/input/RunOnKeyAttribute.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+
+public class RunOnKeyAttribute : RunOnInputAttribute
+{
+    private IInputReceiver inputReceiver;
+
+    public RunOnKeyAttribute(string inputString, InputType type)
+    {
+        InputString = inputString;
+        Type = type;
+    }
+
+    public enum InputType
```

Comment:
I think you should put this in the global namespace to reduce the amount of needed typing when specifying keys.